[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58608 [PyTorch] Rename TI::add_borrowed_{in,out}put to TI::add_{in,out}put
* **#58607 [PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator**
* #58606 [PyTorch] Add TI::add_owned_{in,out}put for clarity & use them

Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

swolchok committed:
[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator

Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

ezyang commented:
btw DISALLOW_TEMPORARIES is defined in a header file so you should probably prefix it with TORCH or something

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

bhosmer reviewed (approved):
LG, echoing @ezyang comment re TORCH prefix

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]

swolchok committed:
Update on "[PyTorch] Add temporary guardrail to borrowing_ op variants on TensorIterator"


Don't let code that tries to pass temporaries to these variants compile.

Differential Revision: [D28524227](https://our.internmc.facebook.com/intern/diff/D28524227/)

[ghstack-poisoned]
